internal/executor/service/job_lease.go

```diff
@@ -81,38 +84,148 @@ func (jobLeaseService *JobLeaseService) RequestJobLeases(availableResource *comm
 		Queues:     leasedQueueReports,
 	}
 
-	leaseRequest := api.LeaseRequest{
+	// Setup a bidirectional gRPC stream.
+	// The server sends jobs over this stream.
+	// The executor sends back acks to indicate which jobs were successfully received.
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
10 seconds here is unfortunately quite short for some cases. I think it should be larger for now until scheduling is changed on the server side
 - It used to be 30s